src/site/topics/testing/unit-tests.rst

```diff
@@ -0,0 +1,151 @@
+
+    .. figure:: test_folder.png
+        :width: 600 px
+        :align: center
+
+        Right click the project in the project view and select New -> Directory. Call this new directory "test".
+
+
+#. Right click the ``test`` folder and select *Mark Directory as -> Test Sources Root*
+
+    .. figure:: mark_tests.png
+        :width: 600 px
+        :align: center
+
+        Right click the new "test" folder to mark it as the "Test Sources Root".
+
+
+#. Create a new class for testing named such that it is the class name followed by "Test"
+
+    * For example, ``FriendTest``
+
+
+#. Add JUnit to the class path
+
+    * A simple way to do it is to write ``@Test`` in the new class and press Alt + Enter
+
+
+#. Select *Add 'JUnit5....*
+
+    .. figure:: alt_enter.png
+        :width: 600 px
+        :align: center
+
+        Pressing Alt + Enter with ``@Test`` selected will cause a popup to appear where "Junit5..." will be an option.
+        Alternatively, one could select the dropdown arrow on the red light bulb for the same menu.
+
+
+Testing the Friend Class
+========================
+
+* Checkout the :download:`FriendTest.java</../test/java/FriendTest.java>` for testing the ``Friend`` class
+* For now, the new ideas introduced here are
+
+    * ``@Test`` annotation --- marking a method as a test
+    * Assertions --- a mechanism for checking something
+
+
+* Below are excerpts from the ``FriendTest`` class
+
+.. literalinclude:: /../test/java/FriendTest.java
+    :language: java
+    :linenos:
+    :lineno-start: 9
+    :lines: 9-13
+
+
+* Consider the first test within the test class as shown above
+* The method is marked with ``@Test`` to mark the method as a test method for Junit
+* It has a descriptive name of the form ``unitUnderTest_case_expectedResult``
+
+    * ``getFirstName_generalCase_returnsFirstName``
+    * ``getFirstName`` is the method being tested
+    * Since there is no real special condition or case for this test, it has ``generalCase`` in its name
+    * ``returnsFirstName`` is the expected result of calling the method
+
+* It uses ``assertEquals`` to check that the expected string ``"qwerty"`` is what is returned by ``getFirstName()``
+
+
+.. literalinclude:: /../test/java/FriendTest.java
+    :language: java
+    :linenos:
+    :lineno-start: 27
+    :lines: 27-39
+
+
+* Consider the above two tests
+
+    * ``equals_equalFriendObjects_areEqual``
+    * ``equals_unequalFriendObjects_areNotEqual``
+
+
+* These tests are named such that the cases (equal/unequal ``Friend`` objects) is clear
+* These tests also make use of different asserts
+
+    * ``assertEquals``
+    * ``assertNotEquals``
+
+
+.. warning::
+
+    The above tests for equality are sufficient for the purposes of this course, but they could be improved. Within the
+    provided ``FriendTest`` class is a test method that makes use of ``EqualsVerifier``, which helps with testing
```

Comment:
May be worth noting that equals verifier is an external dependency not including with java or intellij out of the box.